app/components/timeline_entry/component.rb

```diff
@@ -0,0 +1,65 @@
+module TimelineEntry
+  class Component < ViewComponent::Base
+    def initialize(timeline_event:)
+      super
+      @timeline_event = timeline_event
+    end
+
+    attr_reader :timeline_event
+
+    def description_vars
+      return state_change_vars if timeline_event.state_changed?
+      if timeline_event.reviewer_assigned? || timeline_event.assessor_assigned?
+        return assignment_vars
+      end
+      if timeline_event.assessment_section_completed?
+        assessment_section_completed_vars
+      end
+    end
```

Comment:
I wonder if this would read nicer as a `case`/`when` or even something like `send("#{timeline_event.event_type}_vars")`. I can reasons not to do that though, so again don't feel very strongly on this.

Reply:
eeuw... prettier has done that. It looked much nicer my way 😆